src/pyflame.cc

```diff
@@ -139,24 +194,16 @@ int main(int argc, char **argv) {
         std::this_thread::sleep_for(interval);
         PtraceAttach(pid);
       }
-      if (idle) {
-        std::cout << "(idle) " << idle << "\n";
-      }
-      // process the frames
-      for (const auto &kv : buckets) {
-        if (kv.first.empty()) {
-          std::cerr << "uh oh\n";
+      if (!include_ts) {
+        if (0 != print_frames(idle, buckets)) {
           return 1;
         }
-        auto last = kv.first.rend();
-        last--;
-        for (auto it = kv.first.rbegin(); it != last; ++it) {
-          std::cout << *it << ";";
-        }
-        std::cout << *last << " " << kv.second << "\n";
       }
     } else {
       const unsigned long frame_addr = FirstFrameAddr(pid, tstate_addr);
+      if (include_ts) {
```

Comment:
I initially started of as using a vector to store the call stacks but then after i spoke with you I changed to printing the timestamp callstacks immediately since 
1. C++ prints them fast (also we use \n which does not flush)
2. Don't need to store call stacks in memory for flame chart use case
I can store everything in a vector, the only thing is we will use memory (stack) which probably should be OK as long as the duration of pyflame collection is few mins.
I will consolidate everything like @thepaulm suggested, just that all the call stacks will be printed in similar fashion. ";" separates the call stack function calls, currently if it is a single callstack then we use "\n" to seperate the function calls. That will be the output behavior change for a single call stack